gstack-buffer/src/lib.rs

```diff
@@ -0,0 +1,94 @@
+// This file is part of Gear.
+
+// Copyright (C) 2021-2023 Gear Technologies Inc.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+//! Stack allocations utils.
+
+#![feature(c_unwind)]
+#![no_std]
+
+extern crate alloc;
+
+use alloc::vec::Vec;
+
+use core::{
+    ffi::c_void,
+    mem::{ManuallyDrop, MaybeUninit},
+    slice,
+};
+
+const MAX_BUFFER_SIZE: usize = 64 * 1024;
+
+type Callback = unsafe extern "C-unwind" fn(ptr: *mut MaybeUninit<u8>, data: *mut c_void);
```

Comment:
```suggestion
type Callback = extern "C-unwind" fn(ptr: *mut MaybeUninit<u8>, data: *mut c_void);
```

Reply:
I don't agree with this. If we have C code, then its arguments must also be compatible with the C ABI. So the callback should be marked as unsafe
```c
void f(void(*callback)) { }
```

Reply:
I also checked it with https://github.com/rust-lang/rust-bindgen
This CLI tool also shows the `unsafe` modifier in the callback.
```
$ bindgen alloca_.h

...

extern "C" {
    pub fn c_with_alloca(
        size: usize,
        callback: ::std::option::Option<
            unsafe extern "C" fn(arg1: *mut u8, arg2: *mut ::std::os::raw::c_void),
        >,
        data: *mut ::std::os::raw::c_void,
    );
}
```